driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -1366,15 +1368,18 @@ public Integer getMinConnectionPoolSize() {
     /**
      * Gets the maximum connection pool size specified in the connection string.
      * @return the maximum connection pool size
+     * @see ConnectionPoolSettings#getMaxSize()
      */
     @Nullable
     public Integer getMaxConnectionPoolSize() {
         return maxConnectionPoolSize;
     }
 
     /**
-     * Gets the maximum wait time of a thread waiting for a connection specified in the connection string.
-     * @return the maximum wait time of a thread waiting for a connection
+     * The maximum duration to wait for either an available connection (limited by {@link #getMaxConnectionPoolSize()}),
+     * or a {@linkplain ConnectionCreatedEvent newly created connection} (limited by {@link #getMaxConnecting()}).
```

Comment:
This suggesting contradicts the suggestion made for `ConnectionPoolSettings#getMaxWaitTime`, though both methods have the same meaning. I made no new changes here.

> **Connections become available** when an in-use connection is checked back in to the pool ... or **when a pending connection is added to the pool**

According to the [CMAP spec](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst):

> "available": The Connection has been established and is waiting in the pool to be checked out.

> A Connection MUST NOT be marked as "available" until it has been established.

Thus, when a connection is created, its initial state is "pending", not "available".

Reply:
I am unsure what the contradiction is (I did not endorse the first two lines in the other comment, I only included them for context, in case you are referring to this). Otherwise, I agree, but I am not suggesting that a pending connection should be called available. The original text said:

> Gets the maximum [time to wait] for a connection

The new text says:

> The maximum duration to wait for either an **available** connection or a **newly created connection** [that is, pending connection]

But then, we should either say "to wait for an **available** connection" or "to wait for an **in-use** or **pending** connection to become **available**" (terms bolded for clarity).

Reply:
> I am unsure what the contradiction is

A connection does not "become available ... when a pending connection is added to the pool". This statement (I highlighted it in bold in the previous comment) contradicts the spec.

> The original text said:
> > Gets the maximum [time to wait] for a connection

The original (before the PR) documentation for the `ConnectionString.getMaxWaitTime` method said _"Gets the maximum wait time of a thread waiting for a connection"_. It did not correctly convey the meaning of the `waitQueueTimeoutMS` connection string option.

> But then, we should either say "to wait for an available connection" or "to wait for an in-use or pending connection to become available"

Both of these wordings are incorrect:

- _"to wait for an available connection"_ is incorrect because `waitQueueTimeoutMS` is the max time to wait for either an "available" or a new (a.k.a. "pending") connection;
- _"to wait for an in-use or pending connection to become available"_ is incorrect because `waitQueueTimeoutMS` does not include waiting while a new (a.k.a. "pending") connection becomes "available" (in other words, it does not include the time waiting while a new connection is being established)

Reply:
I updated the `ConnectionString`/`ConnectionPoolSettings.getMaxWaitTime` docs, as well as docs on connection events (the latter document connection states).

Reply:
Nice